x-pack/plugins/apm/server/routes/services/get_service_instances/get_service_instances_transaction_statistics.ts

```diff
@@ -100,13 +119,32 @@ export async function getServiceInstancesTransactionStatistics<
 
   const subAggs = {
     ...getLatencyAggregation(latencyAggregationType, field),
-    failures: {
-      filter: {
-        term: {
-          [EVENT_OUTCOME]: EventOutcome.failure,
+    eventOutcomes: {
+      terms: { field: EVENT_OUTCOME },
+    },
+    errorRate: {
+      bucket_script: {
+        gap_policy: 'insert_zeros' as estypes.AggregationsGapPolicy,
+        buckets_path: {
+          success: "eventOutcomes['success']._count",
+          failure: "eventOutcomes['failure']._count",
         },
+        script: `
+          def success = params.success ?: 0;
+          def failure = params.failure ?: 0;
+          return (success + failure) > 0 ? failure / (success + failure) : 0;
+        `,
       },
     },
+    ...(sortField === 'errorRate'
+      ? {
+          errorRateSort: {
+            bucket_sort: {
+              sort: [{ errorRate: { order: sortDirection } }],
+            },
```

Comment:
I think this will only sort the items returned after the terms aggs run. Is it ok in this case we're trying to fix?

Reply:
yes, that's a problem indeed... just simulated it... thanks for that!

Reply:
I agree it's not ideal. But better than nothing I suppose?

Reply:
... unless we decide to remove the ability to sort by error rate entirely

Reply:
Yeah, that's what we can do with ES at the moment. However, bucket sorting won't probably be any different from sorting directly on the frontend.